python/vision_explanation_methods/error_labeling/error_labeling.py

```diff
@@ -0,0 +1,229 @@
+
+    Only supported for object detection at this point.
+    """
+
+    def __init__(self,
+                 task_type: str,
+                 pred_y: str,
+                 true_y: str,
+                 iou_threshold: float = 0.5):
+        """Create an ErrorLabeling object.
+
+        :param model: The model to explain.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output, with an additional label column.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param task_type: The task to run.
+        :type task_type: str
+        :param classes: Class names as a list of strings.
+            The order of the class names should match that of the model
+            output. Only required if explaining classifier.
+        :type classes: list
+        :param image_mode: The mode to open the image in.
+            See pillow documentation for all modes:
+            https://pillow.readthedocs.io/en/stable/handbook/concepts.html
+        :type image_mode: str
+        """
+        self._is_run = False
+        self._is_added = False
+        self._task_type = task_type
+        self._pred_y = pred_y
+        self._true_y = true_y
+        self._iou_threshold = iou_threshold
+        self._match_matrix = np.full((len(self._true_y), len(self._pred_y)),
+                                     None)
+
+    def compute(self, **kwargs):
+        """Compute the error analysis data.
+
+        Note: if a row does not have a match, that means that there is a
+        missing gt detection
+
+        :param kwargs: The keyword arguments to pass to the compute method.
+            Note that this method does not take any arguments currently.
+        :type kwargs: dict
+        """
+        # save original ordering of predictions
+        original_indices = [i for i, _ in sorted(enumerate(self._pred_y),
+                                                 key=lambda x: x[1][-1],
+                                                 reverse=True)]
+
+        # sort predictions by decreasing conf score
+        # this is to stay consistent with NMS and MAP algorithms
+        sorted_list = sorted(self._pred_y, key=lambda x: x[-1], reverse=True)
+
+        if self._true_y == []:
+            self._match_matrix = [[ErrorLabelType.BACKGROUND]
+                                  for i in range(0, len(self._pred_y))]
+            return
+
+        for gt_index, gt in enumerate(self._true_y):
+            for detect_index, detect in enumerate(sorted_list):
+                iou_score = torchvision.ops.box_iou(
+                    Tensor(detect[1:5]).unsqueeze(0).view(-1, 4),
```

Comment:
Curious why we needed `view(-1, 4)`? Perhaps can mention in comments.

Another nit could be to put `.unsqueeze(0).view(-1, 4))` in a lambda function to reuse considering the particular sequence and values here.

Reply:
We only need the items at index 1 - 4 because that is the location of the bounding box. Will add clarification to the initialization doc string!

Also  - I really like the idea of adding a lambda function! I tried making that change, but looks like that is against flake8 convention (and I think having a separate function def might unnecessarily verbose.